narratives/napoleon/input.py

```diff
@@ -0,0 +1,219 @@
+from datetime import datetime
+import os
+import csv
+import re
+import requests
+
+
+def populate_cd(wid, is_battle):
+    """
+    Populates CachedData table for missing values
+    Returns: ID for new item
+    """
+    URL = "https://query.wikidata.org/sparql"
+    QUERY = """
+    SELECT ?event ?start_time ?point_in_time ?coors
+    WHERE
+    {{
+        BIND(wd:Q{} as ?event)
+        ?event wdt:P276 ?location.
+        ?location wdt:P625 ?coors
+        OPTIONAL {{ ?event wdt:P580 ?start_time. }}
+        OPTIONAL {{ ?event wdt:P585 ?point_in_time. }}
+        SERVICE wikibase:label {{ bd:serviceParam wikibase:language "en" }}
+    }}
+    """.format(
+        wid
+    )
+    event_info = requests.get(URL, params={"format": "json", "query": QUERY}).json()[
+        "results"
+    ]["bindings"]
+    if event_info and "start_time" in event_info and "point_in_time" in event_info:
+        event_info = event_info[0]
+    else:
+        return None
+    cd_date = datetime.strptime(
+        event_info["point_in_time"]["value"][:-1]
+        if "point_in_time" in event_info
+        else event_info["start_time"]["value"][:-1],
+        "%Y-%m-%dT%H:%M:%S",
+    ).strftime("%Y-%m-%d")
+    cd_data = {
+        "event_type": 178_561 if is_battle else 131_569,
+        "wikidata_id": wid,
+        "location": event_info["coors"]["value"],
+        "date": cd_date,
+    }
+    new_cd = requests.post(
+        os.getenv("API_ROOT", "http://localhost/api/") + "/cached-data/",
+        cd_data,
+        params={"format": "json"},
+    )
+    return new_cd.json()["id"]
+
+
+narrative_data = {
+    "author": "Bamber Gascoigne <http://www.historyworld.net>",
+    "title": "Napoleonic Wars",
+    "description": "History of the Napoleonic Wars",
+    "tags": [
+        "military",
+        "conflict",
+        "france",
+        "europe",
+        "revolution",
+        "napoleon",
+        "late modern era",
+    ],
+}
+r_narrative = requests.post(
+    os.getenv("API_ROOT", "http://localhost/api/") + "/narratives/",
+    narrative_data,
+    params={"format": "json"},
+)
+narrative_id = r_narrative.json()["id"]
+print(
+    "Narrative #{}// {}: {}".format(
+        narrative_id, str(r_narrative.status_code), r_narrative.reason
+    )
+)
+
+with open("data.csv", mode="r") as narrative_file:
+    narrative_reader = csv.DictReader(narrative_file)
+    for row in narrative_reader:
+        # TODO: fix placeholder MapSettings
+        ms_data = {
+            "bbox": "MULTIPOINT(-15.751 35.382, 51.046 62.222)",
+            "zoom_min": 1,
+            "zoom_max": 15,
```

Comment:
I think that is too much.

Reply:
Completely agree, like you said I expect these values to be tweaked later